File: DataUtilities_python/Find_Unit.py

def Find_Unit(xds, unit_name):
        
    #%% Find the unit of interest
    
    if unit_name != 'NaN':
        if not isinstance(unit_name, str):
            try:
                # Load the excel file
                from Find_Excel import Find_Excel
                xds_output = Find_Excel(xds)
                # Identify the index of the unit
                unit = xds_output.unit_names[unit_name - 1]
                N = xds.unit_names.index(unit)
            except KeyError:
                N = 'NaN'
        else:
            try:
                N = xds.unit_names.index(unit_name)
            except KeyError:
                N = 'NaN'
    else:
        N = 'NaN'
        
    return N
                















    
    
    
    

File: DataUtilities_matlab/Find_Unit.m
function [N] = Find_Unit(xds, unit_name)

%% Find the unit of interest
if ~ischar(unit_name)

    try
        % Load the excel file
        [xds_output] = Find_Excel(xds);
        unit = xds_output.unit_names(unit_name);
        % Identify the index of the unit
        N = find(strcmp(xds.unit_names, unit));
    catch
        if isnan(unit_name)
            N = [];
        else
            N = unit_name;
        end
    end

else
    N = find(strcmp(xds.unit_names, unit_name));
end

